frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx

```diff
@@ -68,28 +63,34 @@ function isOperatorListItem(item: ListItem): item is OperatorListItem {
 export function AggregationPicker({
   className,
   query,
+  clause,
   stageIndex,
   operators,
   hasExpressionInput = true,
-  legacyQuery,
-  legacyClause,
   maxHeight = DEFAULT_MAX_HEIGHT,
   onSelect,
-  onSelectLegacy,
   onClose,
 }: AggregationPickerProps) {
+  const metadata = useSelector(getMetadata);
+  const displayInfo = clause
+    ? Lib.displayInfo(query, stageIndex, clause)
+    : undefined;
+  const initialOperator = getInitialOperator(query, stageIndex, operators);
   const [
     isEditingExpression,
     { turnOn: openExpressionEditor, turnOff: closeExpressionEditor },
-  ] = useToggle(isExpressionEditorInitiallyOpen(legacyClause));
+  ] = useToggle(isExpressionEditorInitiallyOpen(displayInfo, initialOperator));
```

Comment:
@kamilmielnik @ranquild I understand we use `Lib` inside componetns, but what do you think about some separation? E.g. react components can have access to query and stageIndex, what if we always have a function/util that accepts query and stageIndex and hides the details under the hood. 
it looks like such functions can be tested easily without react involnment.

Reply:
I don't understand, can you please rephrase or show some sample code?